File: 02-Two-Pointers/1768. Merge Strings Alternately/1768. Merge Strings Alternately.cpp
1class Solution {
2public:
3    string mergeAlternately(string word1, string word2) {
4        int n1 = word1.size();
5        int n2 = word2.size();
6        string ans;
7        for (int i = 0; i < max(n1, n2); i++) {
8            if (i < n1)
9                ans += word1[i];
10            if (i < n2)
11                ans += word2[i];
12        }
13
14        return ans;
15    }
16};

File: 01-Arrays-and-Hashing/1768. Merge Strings Alternately/1768. Merge Strings Alternately.java
1class Solution {
2    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
3        // kids    [1,2,3,4,5,6,...]
4        // candies [1,2,4,8,1,3,...]
5        // ExtraCandies...
6        int highestCandies = Arrays.stream(candies).max().getAsInt();
7
8        List<Boolean> ans = new ArrayList<>();
9
10        for (int i = 0; i < candies.length; i++) {
11            ans.add(candies[i] + extraCandies >= highestCandies);
12        }
13
14        return ans;
15    }
16}